main/tests/Makefile.am

```
@@ -69,6 +72,15 @@ test:
 		fi; \
 	fi
 
+uitest:
+	@if test -n "$(assembly)"; then \
+		for asm in $(UITEST_ASSEMBLIES); do \
+			if test `basename $$asm` = $(assembly); then \
+				($(RUN_TEST) -xml=TestResult_`basename $$asm`.xml -labels $$asm) || exit $?; \
+			fi; \
+		done; \
+	fi
```

Okay then, you should fail the test if we're passing an invalid assembly. :P